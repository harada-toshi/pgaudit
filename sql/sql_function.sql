\! cp sql/sql_function/sql_function.conf /tmp/config.conf
\! echo "pgaudit.config_file = '/tmp/config.conf'" >> ./tmp_check/data/postgresql.conf
\! rm ./tmp_check/tmp_check.log
\! pg_ctl restart -w -D ./tmp_check/data -l ./tmp_check/tmp_check.log
\! psql contrib_regression -q -a < sql/sql_function/sql_function.sql
\! rm /tmp/config.conf
\! grep "AUDIT:" ./tmp_check/tmp_check.log > results/sql_function.out

