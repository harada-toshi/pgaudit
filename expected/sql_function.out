	        format = 'AUDIT: %class,%command_tag,%object_type,%object_id,%command_text'
LOG:  AUDIT: DDL,CREATE FUNCTION,FUNCTION,public.test_func(integer),CREATE OR REPLACE FUNCTION test_func(l integer) RETURNS integer AS $$
LOG:  AUDIT: READ,SELECT,,,SELECT test_func(5);
LOG:  AUDIT: FUNCTION,EXECUTE,FUNCTION,public.test_func,SELECT test_func(5);
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: DDL,DROP FUNCTION,FUNCTION,public.test_func(integer),DROP FUNCTION test_func(l integer);
LOG:  AUDIT: FUNCTION,DO,,,DO $$
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
LOG:  AUDIT: READ,SELECT,,,SELECT power(i, 2)
